subprojects/shipkit/src/test/groovy/org/shipkit/internal/gradle/git/GitOriginPluginTest.groovy

```diff
@@ -0,0 +1,11 @@
+package org.shipkit.internal.gradle.git
+
+import testutil.PluginSpecification
+
+class GitOriginPluginTest extends PluginSpecification {
```

Comment:
> Missing test for scenario when plugin is applied but project is not the root.

I know I left a TODO for missing test :) I don't think it is blocking problem for the PR.

>Additionally in GitAuthPluginTest there was a test for provideAuthTo and it wasn't moved here

I must have removed it :) provideAuthTo method does not have any code worth testing. It's all declarative.

We should, however, add an integration test for shipkit init I think. That would be really cool. I wil add a TODO in the code.

Reply:
What about the order (checking if the action provided in argument is actually executed and in the correct moment after repository is already set) and checking if proper repository value is passed to the action?
I think this test here:
https://github.com/mockito/shipkit/pull/424/files#diff-630faa16ee3a0156b085a0e1cc134b61L8
does that. IMHO it's quite simple and does the job. But of course integration test would work too, the question is which is one is simpler/faster/easier to maintain. WDYT?

Reply:
Good feedback! Let's work it out. Here's the implementation suggested in the PR:

```java
        t.dependsOn(identifyTask);
        identifyTask.doLast(new Action<Task>() {
            public void execute(Task task) {
                originRepoName.execute(identifyTask.getRepository());
            }
        });
```

>What about the order (checking if the action provided in argument is actually executed and in the correct moment after repository is already set) and checking if proper repository value is passed to the action?

This seems to me like testing Gradle. I trust that Gradle works:
 - that "doLast" action will get triggered after task is executed
 - that "dependsOn" will ensure correct order

The single line of executing the action and setting the repo is not worth testing - this is just calling a method. It has zero complexity, no conditional logic, no parameter manipulation, no math, no String concat/regexp  - what kind of bug are we guarding against?

Let's role play some "bug scenarios":
 - let's say that during initial implementation, before submitting the PR, I forgot to write "dependsOn" and I did not test this properly e2e. Well if I forget something in the implementation, for sure I would forget it in the unit test, too. Only the integration test would have fleshed this out.
 - let's say that I will plant a bug in the future, I accidentally remove "dependsOn" and it is not detected in code review. Travis CI will fail with downstream tests (which are awesome BTW!!!) because without dependsOn, there is no repo information and stuff breaks.

In summary:
 - in general, we shouldn't write unit tests for a) Gradle functionality b) for code that has zero complexity and is fully "declarative".
 - we are already covered with downstream tests that act as integration tests

The less code, the better, and that includes test code :)

Given that the presence of a unit test is the only contention point and you seem to be happy with the other changes, I will ship this change and take the unit test discussion offline. I'm very happy to converge on the best testing strategy. Please convince me that I need a unit test for this.

Thank you for thoughtful feedback!

Reply:
@mockito/shipkit-developers, guys, can you weigh on this discussion about unit tests? It would be great to hear other opinions so that we can converge on cohesive testing strategy!

Reply:
first of all sorry for the delay, it looks like I somehow overlooked this one.
I do agree with @szczepiq when it comes to testing.  Let's have unit tests for code with high complexity and try to avoid tests for zero complexity/testing gradle functionality.